nemo/collections/nlp/modules/common/megatron/megatron_bert.py

```diff
@@ -151,6 +157,11 @@ def restore_weights(self, restore_path: str):
                 mp_restore_path = f'{restore_path}/mp_rank_{model_parallel_rank:02d}/model_optim_rng.pt'
                 logging.info(f'Restoring model parallel checkpoint from: {mp_restore_path}')
                 state_dict = torch.load(mp_restore_path, map_location='cpu')
+                if 'checkpoint_version' in state_dict:
+                    if state_dict['checkpoint_version'] is not None:
+                        set_checkpoint_version(state_dict['checkpoint_version'])
+                else:
+                    set_checkpoint_version(0)
```

Comment:
Is the `checkpoint_version` added to the state_dict during this call?

Reply:
No, it's not, I'm still trying to figure out how to persist it when we save on the NeMo side.